### ISSUE TITLE
Creation of mandate references relies on unreliable MySQL internals

### ISSUE THREAD

**Issue by rthouvenin** [Project60/org.project60.sepa#609]
When creating a new mandate, the following query is used to guess the next row id and append it to the mandate reference:
`SELECT auto_increment FROM information_schema.tables WHERE table_schema='$database' and table_name='civicrm_sdd_mandate'`.

This is unsafe in several ways:
 - MySQL uses locks for the duration of a **statement** (not transaction) to determine the next auto-increment value, therefore querying the next value from a separate statement does not guarantee it will be the one used by the insert.
 - MySQL has several ways to determine the next value, and in one of them ('interleave') the values are not guaranteed to be sequential. Since MySQL 8.0, this interleave mode is the default (it is more efficient and more reliable)
 - Since MySQL 8.0, by default MySQL caches the table stats in information schema for 24 hours, and the `tables.auto_increment` column is part of these stats. 

As a result, after our migration to MySQL 8 we are not able to create new SEPA mandates because they fail on the unique constraint for the mandate reference...

**Comment by bjendres** [comment]
Yes, thanks @rthouvenin. This code has been in there from the very first prototype, and was always an eyesore :) Would you be able to provide a PR with an alternative?

**Comment by rthouvenin** [comment]
@bjendres Yes will work out something, hopefully very soon as this is quite urgent for us now :)

**Comment by bjendres** [comment]
> Yes will work out something hopefully very soon 

Great!

> ... as this is quite urgent for us now :)

Sorry about that :/

**Comment by MasterEmit** [comment]
This is still open? Just after upgrading everything I got some days later the reply, that our office could sometimes not create sepa mandates and today I realize, that this problem still exists :-(

**Comment by bjendres** [comment]
> This is still open? 

Yes, because of [this](https://github.com/Project60/org.project60.sepa/pull/610#issuecomment-986285789) issue.

> Just after upgrading everything I got some days later the reply, that our office could sometimes not create sepa mandates and today I realize, that this problem still exists :-(

Sorry about that. Would you be able to help out with the PR (#610)?
